cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/service/MetadataHttpHandler.java

```diff
@@ -343,4 +347,21 @@ public void removeStreamTags(HttpRequest request, HttpResponder responder,
   public void recordRun(HttpRequest request, HttpResponder responder) {
     responder.sendString(HttpResponseStatus.OK, "Metadata recorded successfully");
   }
+
+  // *** Search endpoints ***
+
+  @GET
+  @Path("/namespaces/{namespace-id}/search")
+  public void searchMetadata(HttpRequest request, HttpResponder responder,
+                             @PathParam("namespace-id") String namespaceId,
+                             @QueryParam("query") String searchQuery,
+                             @QueryParam("target") String target) throws Exception {
+    if (Strings.isNullOrEmpty(target)) {
+      throw new BadRequestException(String.format("Invalid type for search '%s'", searchQuery));
+    }
+
+    Iterable<MetadataSearchResultRecord> results = metadataAdmin.searchMetadataOnType(searchQuery, target);
```

Comment:
until we finish implementation of searchMetadataOnType, should we send some mock response from this API? Just in case UI wants to integrate early.